jacodb-analysis/src/main/kotlin/org/jacodb/analysis/util/EtsTraits.kt

```diff
@@ -0,0 +1,160 @@
+/*
+ *  Copyright 2022 UnitTestBot contributors (utbot.org)
+ * <p>
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ * <p>
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.jacodb.analysis.util
+
+import org.jacodb.analysis.ifds.AccessPath
+import org.jacodb.analysis.ifds.ElementAccessor
+import org.jacodb.analysis.ifds.FieldAccessor
+import org.jacodb.analysis.util.toPathOrNull
+import org.jacodb.api.common.CommonMethodParameter
+import org.jacodb.api.common.cfg.CommonArgument
+import org.jacodb.api.common.cfg.CommonCallExpr
+import org.jacodb.api.common.cfg.CommonExpr
+import org.jacodb.api.common.cfg.CommonValue
+import org.jacodb.panda.dynamic.ets.base.EtsAnyType
+import org.jacodb.panda.dynamic.ets.base.EtsArrayAccess
+import org.jacodb.panda.dynamic.ets.base.EtsCallExpr
+import org.jacodb.panda.dynamic.ets.base.EtsCastExpr
+import org.jacodb.panda.dynamic.ets.base.EtsConstant
+import org.jacodb.panda.dynamic.ets.base.EtsEntity
+import org.jacodb.panda.dynamic.ets.base.EtsImmediate
+import org.jacodb.panda.dynamic.ets.base.EtsInstanceFieldRef
+import org.jacodb.panda.dynamic.ets.base.EtsParameterRef
+import org.jacodb.panda.dynamic.ets.base.EtsStaticFieldRef
+import org.jacodb.panda.dynamic.ets.base.EtsStmt
+import org.jacodb.panda.dynamic.ets.base.EtsThis
+import org.jacodb.panda.dynamic.ets.base.EtsValue
+import org.jacodb.panda.dynamic.ets.model.EtsMethod
+import org.jacodb.panda.dynamic.ets.model.EtsMethodImpl
+import org.jacodb.panda.dynamic.ets.model.EtsMethodParameter
+import org.jacodb.panda.dynamic.ets.utils.callExpr
+import org.jacodb.taint.configuration.ConstantValue
+import org.jacodb.analysis.util.toPath as _toPath
+import org.jacodb.analysis.util.toPathOrNull as _toPathOrNull
+import org.jacodb.panda.dynamic.ets.utils.getOperands as _getOperands
+import org.jacodb.panda.dynamic.ets.utils.getValues as _getValues
+
+interface EtsTraits : Traits<EtsMethod, EtsStmt> {
+
+    override val CommonCallExpr.callee: EtsMethod
+        get() {
+            check(this is EtsCallExpr)
+            // return cp.getMethodBySignature(method) ?: error("Method not found: $method")
+            return EtsMethodImpl(method)
+        }
+
+    override val EtsMethod.thisInstance: EtsThis
+        get() = EtsThis(EtsAnyType)
+
+    override val EtsMethod.isConstructor: Boolean
+        get() = false
```

Comment:
It was actually a TODO.

I have fixed it to be `name == "constructor"`. We need some tests for it.

Actually, we might just drop `isConstructor` from Traits, since originally it was just a tricky adhoc, and now it seems to be unnecessary...